elements/pfe-clipboard/pfe-clipboard.ts

```diff
@@ -0,0 +1,323 @@
+        }
+        break;
+      }
+    }
+
+    if (!text || (typeof text === 'string' && !text.length)) {
+      this.logger.error('Couldn\'t find text to copy.');
+      this.setAttribute('disabled', '');
+      return;
+    }
+
+    // Execute the copy to clipboard functionality
+    try {
+      const copiedText = await this.copyTextToClipboard(text);
+      // Emit event that lets others know the user has "copied"
+      // the url. We are also going to include the url that was
+      // copied.
+      this.dispatchEvent(pfeEvent('copied', { copiedText }));
+      /**
+       * @deprecated
+       * @fires {CustomEvent<{ component: PfeClipboard }>} pfe-clipboard:connected
+       */
+      this.dispatchEvent(pfeEvent('pfe-clipboard:copied', {
+        url: copiedText, // @todo deprecate
+        copiedText,
+      }));
+      // Toggle the copied state. Use the this._formattedCopiedTimeout function
+      // to an appropraite setTimout length.
+      this.setAttribute('copied', '');
+      setTimeout(() => {
+        this.removeAttribute('copied');
+      }, this._formattedCopiedTimeout());
+    } catch (error) {
+      this.logger.warn(error as string);
+      this._checkForCopyTarget();
+    }
+  }
+
+  protected _contentToCopyChanged() {
+    if (this.contentToCopy)
+      this.removeAttribute('disabled');
+  }
+
+  /**
+   * Formatted copied timeout value. Use the formattedCopiedTimeout function
+   * to get a type safe, millisecond value of the timeout duration.
+   */
+  private _formattedCopiedTimeout() {
+    const copiedDuration = Number(this.copiedDuration * 1000);
+    if (!(copiedDuration > -1)) {
+      this.logger.warn(`copied-duration must be a valid number. Defaulting to 3 seconds.`);
+      // default to 3 seconds
+      return 3000;
+    } else
+      return copiedDuration;
+  }
+
+  /**
+   * Listen for keyboard events and map them to their
+   * corresponding mouse events.
+   */
+  @bound private _keydownHandler(event: KeyboardEvent) {
+    const { key } = event;
+    switch (key) {
+      case 'Enter':
+        this._clickHandler();
+        break;
+      case ' ':
+        // Prevent the browser from scolling when the user hits the space key
+        event.stopPropagation();
+        event.preventDefault();
+        this._clickHandler();
+        break;
+    }
+  }
+
+  /**
+   * Copy arbitrary text to system clipboard
+   *
+   * If available, it will use the [Clipboard API](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/clipboard)
+   * to access the system clipboard
+   *
+   * If Clipboard is unavailable, it will use the legacy [`execCommand`](https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand)
+   *
+   * ```js
+   * document.querySelector("pfe-clipboard").copyTextToClipboard(text)
+   *   .then(res => console.log(`Successfully copied: ${res}`))
+   *   .catch(error => console.error(error));
+   * ```
+   *
+   */
+  async copyTextToClipboard(text: string): Promise<string> {
```

Comment:
Is there a way to include this in CEM?  I think we would want to advertise this as a publicly available function.

Reply:
it is included? https://deploy-preview-1804--patternfly-elements.netlify.app/components/clipboard/#copytexttoclipboardtext-string

Reply:
It's included in the docs section but not in the `api-viewer`.  I didn't know if CEM had the connotation of documenting public APIs.

Reply:
Never mind, didn't realize the docs were automatically generated 🤦   Nice!

Reply:
does api-viewer have methods support?
if we don't already, we should add a parallel analyze watcher to the `start` script

@heyMP what's your opinion on whether-or-not to commit the `custom-elements.json` files?

Reply:
I was thinking that they would be generated at build time and published in the npm packages therefore we would keep them in `.gitignore`.

+1 on the parallel analyze watcher.

Reply:
Added it as an issue.

Reply:
ok I implemented this just now